plans/objchange/objchange.go

```diff
@@ -118,165 +122,169 @@ func proposedNewObject(schema *configschema.Block, prior, config cty.Value) cty.
 	for name, blockType := range schema.BlockTypes {
 		priorV := prior.GetAttr(name)
 		configV := config.GetAttr(name)
-		var newV cty.Value
-		switch blockType.Nesting {
+		newAttrs[name] = proposedNewNestedBlock(blockType, priorV, configV)
+	}
 
-		case configschema.NestingSingle, configschema.NestingGroup:
-			newV = ProposedNewObject(&blockType.Block, priorV, configV)
+	return cty.ObjectVal(newAttrs)
+}
+
+func proposedNewNestedBlock(schema *configschema.NestedBlock, prior, config cty.Value) cty.Value {
+	var newV cty.Value
+
+	switch schema.Nesting {
+
+	case configschema.NestingSingle, configschema.NestingGroup:
+		newV = ProposedNewObject(&schema.Block, prior, config)
+
+	case configschema.NestingList:
+		// Nested blocks are correlated by index.
+		configVLen := 0
+		if config.IsKnown() && !config.IsNull() {
+			configVLen = config.LengthInt()
+		}
+		if configVLen > 0 {
+			newVals := make([]cty.Value, 0, configVLen)
+			for it := config.ElementIterator(); it.Next(); {
+				idx, configEV := it.Element()
+				if prior.IsKnown() && (prior.IsNull() || !prior.HasIndex(idx).True()) {
+					// If there is no corresponding prior element then
+					// we just take the config value as-is.
+					newVals = append(newVals, configEV)
+					continue
+				}
+				priorEV := prior.Index(idx)
 
-		case configschema.NestingList:
-			// Nested blocks are correlated by index.
+				newEV := ProposedNewObject(&schema.Block, priorEV, configEV)
+				newVals = append(newVals, newEV)
+			}
+			// Despite the name, a NestingList might also be a tuple, if
+			// its nested schema contains dynamically-typed attributes.
+			if config.Type().IsTupleType() {
+				newV = cty.TupleVal(newVals)
+			} else {
+				newV = cty.ListVal(newVals)
+			}
+		} else {
+			// Despite the name, a NestingList might also be a tuple, if
+			// its nested schema contains dynamically-typed attributes.
+			if config.Type().IsTupleType() {
+				newV = cty.EmptyTupleVal
+			} else {
+				newV = cty.ListValEmpty(schema.ImpliedType())
```

Comment:
An attribute would need to h ave `cty.NullVal(schema.ImpliedType())` here, while a block uses an empty value